addons/project/data/project_demo.xml

```diff
@@ -1320,6 +1321,79 @@ Send it ASAP, its urgent.</field>
+            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
+        </function>
+        <function model="mail.message" name="write">
+            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_1'))]"/>
+            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
+        </function>
+        <function model="mail.message" name="write">
+            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_2'))]"/>
+            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
+        </function>
+        <function model="mail.message" name="write">
+            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_3'))]"/>
+            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
+        </function>
+        <function model="mail.message" name="write">
+            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_4'))]"/>
+            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
+        </function>
+        <function model="mail.message" name="write">
+            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_5'))]"/>
+            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
+        </function>
+        <function model="mail.message" name="write">
+            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_6'))]"/>
+            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
+        </function>
+        <function model="mail.message" name="write">
+            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_7'))]"/>
+            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
+        </function>
+        <function model="mail.message" name="write">
+            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_8'))]"/>
+            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
+        </function>
+        <function model="mail.message" name="write">
+            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_3_task_2'))]"/>
+            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
+        </function>
```

Comment:
Can you do it in batch? 

```xml
        <function model="mail.message" name="write">
            <value model="mail.message" search="[
                ('subtype_id', '=', ref('project.mt_task_new')), 
                ('res_id', 'in', [
                    ref('project.project_1_task_1'),
                    ref('project.project_1_task_2'),
                    ...
                ])
            ]"/>
            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
        </function>
```

Reply:
Hello @xavierbol 

I've Tried to do it in batch but didn't work, the write method was only called for the last res_id
I think we couldn't do it in batch.

Reply:
Indeed, the orm takes the first id found with the search but we can do it with the `eval` attribute. 🙂 

Here is the diff:

```diff
diff --git a/addons/project/data/project_demo.xml b/addons/project/data/project_demo.xml
index b660c0adead9..b28b7bc666fa 100644
--- a/addons/project/data/project_demo.xml
+++ b/addons/project/data/project_demo.xml
@@ -1323,75 +1323,31 @@ Send it ASAP, its urgent.</field>
 
         <!-- Change task creation notifications date -->
         <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_1_task_1'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_1_task_2'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_1_task_3'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_1_task_4'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_1_task_5'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_1_task_6'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_1_task_7'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_1_task_8'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_1_task_9'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_1'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_2'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_3'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_4'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_5'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_6'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_7'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_2_task_8'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('project.project_3_task_2'))]"/>
+            <value model="mail.message"
+                   eval="obj().env['mail.message'].search([
+                            ('subtype_id', '=', ref('project.mt_task_new')),
+                            ('res_id', 'in', [
+                                ref('project.project_1_task_1'),
+                                ref('project.project_1_task_2'),
+                                ref('project.project_1_task_3'),
+                                ref('project.project_1_task_4'),
+                                ref('project.project_1_task_5'),
+                                ref('project.project_1_task_6'),
+                                ref('project.project_1_task_7'),
+                                ref('project.project_1_task_8'),
+                                ref('project.project_1_task_9'),
+                                ref('project.project_2_task_1'),
+                                ref('project.project_2_task_2'),
+                                ref('project.project_2_task_3'),
+                                ref('project.project_2_task_4'),
+                                ref('project.project_2_task_5'),
+                                ref('project.project_2_task_6'),
+                                ref('project.project_2_task_7'),
+                                ref('project.project_2_task_8'),
+                                ref('project.project_3_task_2'),
+                            ]),
+                        ]).ids"
+            />
             <value eval="{'date': DateTime.now() - relativedelta(months=5)}"/>
         </function>
 
diff --git a/addons/sale_timesheet/data/sale_service_demo.xml b/addons/sale_timesheet/data/sale_service_demo.xml
index 3850007d3c8d..10568ef3b809 100644
--- a/addons/sale_timesheet/data/sale_service_demo.xml
+++ b/addons/sale_timesheet/data/sale_service_demo.xml
@@ -2583,11 +2583,12 @@
 
         <!-- Change task creation notifications date -->
         <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('sale_timesheet.project_dpc_task_4'))]"/>
-            <value eval="{'date': DateTime.now() - relativedelta(days=4)}"/>
-        </function>
-        <function model="mail.message" name="write">
-            <value model="mail.message" search="[('subtype_id', '=', ref('project.mt_task_new')), ('res_id', '=', ref('sale_timesheet.project_dpc_task_5'))]"/>
+            <value model="mail.message"
+                   eval="obj().env['mail.message'].search([
+                        ('subtype_id', '=', ref('project.mt_task_new')),
+                        ('res_id', 'in', [ref('sale_timesheet.project_dpc_task_4'), ref('sale_timesheet.project_dpc_task_5')]),
+                    ]).ids"
+            />
             <value eval="{'date': DateTime.now() - relativedelta(days=4)}"/>
         </function>
     </data>
```